pg_search/src/postgres/customscan/pdbscan/pushdown.rs

```diff
@@ -97,7 +97,17 @@ pub unsafe fn try_pushdown(
     schema: &SearchIndexSchema
 ) -> Option<Qual> {
     let args = PgList::<pg_sys::Node>::from_pg((*opexpr).args);
-    let var = nodecast!(Var, T_Var, args.get_ptr(0)?)?;
+    let var = {
+        // inspect the left-hand-side of the operator expression...
+        let mut lhs = args.get_ptr(0)?;
+        
+        while (*lhs).type_ == pg_sys::NodeTag::T_RelabelType {
+            // and keep following it as long as it's still a RelabelType
+            let relabel_type = lhs as *mut pg_sys::RelabelType;
+            lhs = (*relabel_type).arg as _;
+        }
+        nodecast!(Var, T_Var, lhs)?
+    };
```

Comment:
This code only handles `RelabelType` on the left-hand side of expressions. There may be cases where the right-hand side also has implicit type conversion. Though, I see that in several other places in the code-base, e.g., constants are considered to be on the right side.

Reply:
Yeah, agreed. We can also have a helper rhs/lhs switch early on, to make sure all constants appear on lhs.